lib/net/http/header.rb

```diff
@@ -491,7 +491,7 @@ def each_capitalized
   alias canonical_each each_capitalized
 
   def capitalize(name)
-    name.to_s.split(/-/).map {|s| s.capitalize }.join('-')
+    name.to_s.split('-'.freeze).map {|s| s.capitalize }.join('-'.freeze)
```

Comment:
Not what you are asking for, but another implementation work trying could be:

```ruby
def capitalize(name)
  name.to_s.gsub(/(\A|(?<=[\^\-]))([a-z])/) do |c|
    c.upcase!
    c
  end
end
```

Reply:
I did some benchmarking, and using the regex seemed to be slower.